### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by KorneyKorneich**
Implement email/password signIn/signUp & phone number signIn. Also log out was implemented

### CODE REVIEW

File: src/entities/country/ui/CountryItem.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    handleCountryPick: (country: Country) => void,
+}
+
+export const CountryItem = (props: CountryItemProps) => {
+	const {item, handleCountryPick} = props;
+
+
+	return (
+		<TouchableOpacity onPress={() => handleCountryPick(item)} style={styles.countryContainer}>
+			<Image source={{ uri: item.flag }} resizeMode={"contain"} style={{ width: 40, height: 25 }} />
```
Comment (VladisAga) [inline src/entities/country/ui/CountryItem.tsx]: styles should be in a separate file, and in the shared/ui you can find CustomImage, use it

File: src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			if(selectedCountry){
+				const confirm = await phoneAuth({selectedCountry: selectedCountry, phone: phone});
+				setConfirmation(confirm);
+			}
+			navigation.navigate( "OTPVerifyScreen", { confirmation: confirmation } );
+
+
+
+		}catch (e){
+			console.log(e);
```
Comment (VladisAga) [inline src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx]: in catch should bs smth, but not the console.log. Or you can remove try catch

File: src/shared/ui/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+//consts
+export {StyleGuide} from "./stylesConsts/stylesConsts";
+
+//components
+export {CustomInput} from "./Components/CustomInput/ui/CustomInput";
+export {CustomButton} from "./Components/CustomButton/ui/CustomButton";
+export { Map } from "./Components/Map/ui/Map";
+export {CustomImage} from "./Components/CustomImage/ui/CustomImage";
+export {CustomCheckBox} from "./Components/CustomCheckBox/ui/CustomCheckBox";
```
Comment (VladisAga) [inline src/shared/ui/index.ts:18]: remove comments

File: src/shared/ui/stylesConsts/stylesConsts.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { Dimensions } from "react-native";
+
+export enum StyleGuide {
+    GREY = "#e3e3e3",
+    WHITE ="#FFFFFF",
+    BLACK= "#130F26",
+    PRIMARY ="#33AD5F",
+
+    //TEXT COLORS
```
Comment (VladisAga) [inline src/shared/ui/stylesConsts/stylesConsts.ts:9]: comments

File: src/shared/ui/stylesConsts/stylesConsts.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { Dimensions } from "react-native";
+
+export enum StyleGuide {
+    GREY = "#e3e3e3",
```
Comment (VladisAga) [inline src/shared/ui/stylesConsts/stylesConsts.ts:4]: i suppose you to use rgb or rgba

**Review by VladisAga — CHANGES_REQUESTED** [review]
small changes

File: src/shared/ui/index.ts (hunk 3 above)
Comment (KorneyKorneich) [reply src/shared/ui/index.ts:18]: i think that comments here are acceptable for better structuring

File: src/shared/ui/stylesConsts/stylesConsts.ts (hunk 4 above)
Comment (KorneyKorneich) [reply src/shared/ui/stylesConsts/stylesConsts.ts:9]: same

**Review by VladisAga — APPROVED** [review]

**Pull request merged**